### ISSUE TITLE
Duplicate JPG photos appear

### ISSUE THREAD

**Issue by 404gods** [boredazfcuk/docker-icloudpd#163]
Hello, I am using Synology NAS, I used the variable set_exif_datetime:True in order to display the shooting time in JPG format pictures, everything looks normal.

However, I found that some duplicate JPG images appeared, with the same file size and the same content, named IMG_1234-12345.jpg These images all had EXIF information with the correct capture time.

And, the log content prompts as follows:
2022-05-24 15:06:22 DEBUG Setting EXIF timestamp for /photo/icloud/2021/IMG_5701.JPG: 2021-12-15 10:13:00+0800
2022-05-24 15:06:28 INFO /photo/icloud/2021/IMG_5701-435527.JPG deduplicated.
2022-05-24 15:06:28 INFO Downloading /photo/icloud/2021/IMG_5701-435527.JPG
2022-05-24 15:06:28 DEBUG Setting EXIF timestamp for /photo/icloud/2021/IMG_5701-435527.JPG: 2021-12-15 10:13:00+0800

![屏幕截图 2022-05-24 151912](https://user-images.githubusercontent.com/50573302/169975874-f154abdf-874f-44eb-af68-bd4b2ece7fdf.png)
How can I avoid these duplicate jpg files? thanks

**Comment by boredazfcuk** [comment]
Hi,

This is a result of the `set_exif_datetime` variable being set to `True`.

The icloud_photos_downloader application has de-duplication built into it as it's possible to have two identically named files exist in your iCloud photo stream.

When the downloader runs, if it finds a file on the local disk drive which does not match the file in iCloud, it saves the file by appending some digits derived from the file size to the filename, to make it individual.

The issue is occurring because your photo in iCloud does not have the EXIF data saved into it, but the file on the disk does. This means they are differnet files, so icloud_photos_downloader saves the original from iCloud with the appended file name.

I raised this as an issue in the repository for icloud_photo_downloader in September, as I believe this behaviour is unwanted, but no pogress has been since then. You can track the issue here: https://github.com/icloud-photos-downloader/icloud_photos_downloader/issues/380

**Comment by 404gods** [comment]
> then

This is really troublesome. Photos in jpg format taken by third-party apps will lose EXIF ​​information when uploaded to icloud.  After adding variables, duplicate pictures will be generated. These two problems are completely unbearable for managing albums.

**Comment by boredazfcuk** [comment]
What you mean 'lose' EXIF data? The photos shouldn't lose anything? The `set_exif_datetime` variable exists because some photos are missing EXIF data, so this variable will create the EXIF information from the timestamp of the file.

But I agree that it's not fit for purpose. I ended up switching it off in the end. When I examined my photostream, I saw that the only photos which were being duplicated to to missing EXIF data were images I'd downloaded in Safari, or from Reddit, or saved from chat apps such as Signal, WhatsApp and Telegram. I just stick to taking photos through the Camera app now as I don't really have a problem is those types of photos don't have EXIF data.
